coherent_rates/plot.py

```diff
@@ -507,3 +513,111 @@ def plot_scattered_energy_change_comparison(
     fig.legend()
     fig.show()
     input()
+
+
+def plot_scattered_energy_change_state(
+    system: PeriodicSystem,
+    config: PeriodicSystemConfig,
+    state: StateVector[Any],
+    *,
+    nk_points: list[tuple[int, ...]] | None = None,
+) -> None:
+    fig, ax = get_figure(None)
+
+    bound_data = get_scattered_energy_change_against_k(
+        system,
+        config,
+        state,
+        nk_points=nk_points,
+    )
+
+    k_points = bound_data["basis"].k_points
+    k_points_squared = np.square(k_points)
+
+    ax.plot(k_points_squared, bound_data["data"], label="Quadratic")
+    fig.legend()
+    ax.set_ylabel("Energy change")
+    ax.set_xlabel("k^2")
+    fig.show()
+
+    fig, ax = get_figure(None)
+    ax.plot(k_points, bound_data["data"], label="Linear")
+    ax.set_ylabel("Energy change")
+    ax.set_xlabel("k")
+    fig.legend()
+    fig.show()
+    input()
+
+
+def plot_occupation_against_energy_comparison(
+    system: PeriodicSystem,
+    config: PeriodicSystemConfig,
+    mass_ratio: float,
+    direction: tuple[int, ...] | None = None,
+) -> None:
+    direction = tuple(1 for _ in config.shape) if direction is None else direction
+
+    hamiltonian = get_hamiltonian(system, config)
```

Comment:
should add a generic plot_band_occupation_against_energy which takes a state vector with an ExplicitStackedBasis, and gets the band vs bloch k information from basis.vectors["basis"][0]

Reply:
In this file, not in surface potential analysis